source/core/slang-uint-set.h

```diff
@@ -6,22 +6,48 @@
 #include "slang-common.h"
 #include "slang-hash.h"
 
+#if defined(_MSC_VER)
+#include <intrin.h>
+#endif
 #include <memory.h>
 
 namespace Slang
 {
 
+template<typename T>
+constexpr static Index computeElementShift()
+{
+    Index currentShift = 0;
+    Index currentShiftValue = 1;
+
+    while (currentShiftValue != sizeof(T)*8)
+    {
+        currentShift++;
+        currentShiftValue *= 2;
+    }
+
+    return currentShift;
+}
+
 /* Hold a set of UInt values. Implementation works by storing as a bit per value */
+/// UIntSet is essentially a Element[], where each Element is `b` bits big.
+/// Each index has `b` number of integers. If the bit is 1, we have an element there. 
+/// Value of each element is equal to the binary offset from Element[0], bit 0.
 class UIntSet
 {
 public:
     typedef UIntSet ThisType;
-    typedef uint32_t Element;                                   ///< Type that holds the bits to say if value is present
+    typedef uint64_t Element;                                   ///< Type that holds the bits to say if value is present
```

Comment:
the total complexity of the algorithm for `getElements` is N, where N is the number of set bits (10-20). It is likely quite a bit more expensive for us to add an iterator.

Reply:
Right, there isn't much computational benefits.
The value comes from the removal of additional memory allocation from creating that temp list.

Reply:
I think I will avoid adding it in this PR then (has no use for capabilities since we need all capabilities at once, better to bulk allocate in this scenario).

Reply:
Sure, the PR is good as is.